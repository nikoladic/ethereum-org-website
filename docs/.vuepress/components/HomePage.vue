<template>
  <div>
    <div class="headline-text-container">
      <h1 class="headline-title">{{content['page-home-title']}}</h1>
      <div class="headline-subtitle">{{content['page-home-subtitle']}}</div>
    </div>

    <div class="intro-blocks">
      <router-link to="/beginners/" class="intro-block">
        <h3>
          <span class="arrow">→</span>{{content['page-home-section-beginners-title']}}
        </h3>
        <ul>
          <li class="highlight">{{content['page-home-section-beginners-item-one']}}</li>
          <li>{{content['page-home-section-beginners-item-two']}}</li>
          <li>{{content['page-home-section-beginners-item-three']}}</li>
        </ul>
      </router-link>

      <div class="intro-block">
        <h3>
          <router-link to="/use/">
            <span class="arrow">→</span>{{content['page-home-section-use-title']}}
          </router-link>
        </h3>
        <ul>
          <li>
            <router-link
              to="/use/#_1-use-an-application-built-on-ethereum"
              class="black"
            >{{content['page-home-section-use-item-one']}}</router-link>
          </li>
          <li>
            <router-link
              to="/use/#_2-what-is-eth-and-how-do-i-get-it"
              class="black"
            >{{content['page-home-section-use-item-two']}}</router-link>
          </li>
          <li>
            <router-link
              to="/use/#_3-what-is-a-wallet-and-which-one-should-i-use"
              class="black"
            >{{content['page-home-section-use-item-three']}}</router-link>
          </li>
        </ul>
      </div>

      <div class="intro-block">
        <h3>
          <router-link to="/learn/">
            <span class="arrow">→</span>{{content['page-home-section-learn-title']}}
          </router-link>
        </h3>
        <ul>
          <li>
            <router-link to="/learn/#ethereum-basics" class="black">{{content['page-home-section-learn-item-one']}}</router-link>
          </li>
          <li>
            <router-link to="/learn/#how-ethereum-works" class="black">{{content['page-home-section-learn-item-two']}}</router-link>
          </li>
          <li>
            <router-link to="/learn/#eth-2-0" class="black">{{content['page-home-section-learn-item-three']}}</router-link>
          </li>
        </ul>
      </div>

      <div class="intro-block">
        <h3>
          <router-link to="/developers/">
            <span class="arrow">→</span>{{content['page-home-section-developers-title']}}
          </router-link>
        </h3>
        <ul>
          <li>
            <router-link to="/developers/#getting-started" class="black">{{content['page-home-section-developers-item-one']}}</router-link>
          </li>
          <li>
            <router-link
              to="/developers/#smart-contract-languages"
              class="black"
            >{{content['page-home-section-developers-item-two']}}</router-link>
          </li>
          <li>
            <router-link
              to="/developers/#developer-tools"
              class="black"
            >{{content['page-home-section-developers-item-three']}}</router-link>
          </li>
        </ul>
      </div>

      <div class="intro-block" v-if="shouldShowEnterprise">
        <h3>
          <router-link to="/enterprise/">
            <span class="arrow">→</span>{{content['page-home-section-enterprise-title']}}
          </router-link>
        </h3>
        <ul>
          <li>
            <router-link
              to="/enterprise/#why-enterprise-ethereum"
              class="black"
            >{{content['page-home-section-enterprise-item-one']}}</router-link>
          </li>
          <li>
            <router-link to="/enterprise/#enterprise-features" class="black">{{content['page-home-section-enterprise-item-two']}}</router-link>
          </li>
          <li>
            <router-link
              to="/enterprise/#enterprise-developer-community"
              class="black"
            >{{content['page-home-section-enterprise-item-three']}}</router-link>
          </li>
        </ul>
      </div>
    </div>
  </div>
</template>

<script>
  import { translate } from '../theme/utils/translations'

  export default {
    computed: {
      // TODO there must be a better way...
      // How to use translate() within Vue component template??
      content() {
        return {
          'page-home-title': translate('page-home-title', this.$lang),
          'page-home-subtitle': translate('page-home-subtitle', this.$lang),
          'page-home-section-beginners-title': translate('page-home-section-beginners-title', this.$lang),
          'page-home-section-beginners-item-one': translate('page-home-section-beginners-item-one', this.$lang),
          'page-home-section-beginners-item-two': translate('page-home-section-beginners-item-two', this.$lang),
          'page-home-section-beginners-item-three': translate('page-home-section-beginners-item-three', this.$lang),
          'page-home-section-use-title': translate('page-home-section-use-title', this.$lang),
          'page-home-section-use-item-one': translate('page-home-section-use-item-one', this.$lang),
          'page-home-section-use-item-two': translate('page-home-section-use-item-two', this.$lang),
          'page-home-section-use-item-three': translate('page-home-section-use-item-three', this.$lang),
          'page-home-section-learn-title': translate('page-home-section-learn-title', this.$lang),
          'page-home-section-learn-item-one': translate('page-home-section-learn-item-one', this.$lang),
          'page-home-section-learn-item-two': translate('page-home-section-learn-item-two', this.$lang),
          'page-home-section-learn-item-three': translate('page-home-section-learn-item-three', this.$lang),
          'page-home-section-developers-title': translate('page-home-section-developers-title', this.$lang),
          'page-home-section-developers-item-one': translate('page-home-section-developers-item-one', this.$lang),
          'page-home-section-developers-item-two': translate('page-home-section-developers-item-two', this.$lang),
          'page-home-section-developers-item-three': translate('page-home-section-developers-item-three', this.$lang),
          'page-home-section-enterprise-title': translate('page-home-section-enterprise-title', this.$lang),
          'page-home-section-enterprise-item-one': translate('page-home-section-enterprise-item-one', this.$lang),
          'page-home-section-enterprise-item-two': translate('page-home-section-enterprise-item-two', this.$lang),
          'page-home-section-enterprise-item-three': translate('page-home-section-enterprise-item-three', this.$lang)
        }
      },

      // only English has Enterprise page
      shouldShowEnterprise() {
        return this.$lang === 'en-US'
      }
    }
  }
</script>