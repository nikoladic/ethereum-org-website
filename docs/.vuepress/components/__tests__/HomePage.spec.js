import { shallowMount } from '@vue/test-utils'
import HomePage from "@/components/HomePage"

describe('HomePage', () => {
  test('show enterprise with en-US', () => {
    const wrapper = shallowMount(HomePage, {
      stubs: ['router-link'],
      mocks: {
        $lang: "en-US"
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('render successfully with Deutsch', () => {
    const wrapper = shallowMount(HomePage, {
      stubs: ['router-link'],
      mocks: {
        $lang: "de"
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
