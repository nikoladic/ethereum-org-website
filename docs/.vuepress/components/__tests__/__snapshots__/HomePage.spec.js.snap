// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage render successfully with Deutsch 1`] = `
<div>
  <div
    class="headline-text-container"
  >
    <h1
      class="headline-title"
    >
      Ethereum ist eine globale Open-Source-Plattform für dezentralisierte Anwendungen.
    </h1>
     
    <div
      class="headline-subtitle"
    >
      Auf Ethereum kannst du Code schreiben, der digitale Werte verwaltet, der exakt wie programmiert ausgeführt wird und der von überall auf der Welt zugänglich ist.
    </div>
  </div>
   
  <div
    class="intro-blocks"
  >
    <div
      class="intro-block"
    >
      <h3>
        <router-link-stub
          to="/de/beginners/"
        >
          <span
            class="arrow"
          >
            →
          </span>
          Anfänger
        
        </router-link-stub>
      </h3>
       
      <ul>
        <li
          class="highlight"
        >
          <router-link-stub
            class="black"
            to="/de/beginners/"
          >
            
            Neu bei Ethereum?
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/de/beginners/"
          >
            
            Was ist Ethereum?
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/de/beginners/"
          >
            
            Welchen Nutzen bringt es mir?
          
          </router-link-stub>
        </li>
      </ul>
    </div>
     
    <div
      class="intro-block"
    >
      <h3>
        <router-link-stub
          to="/de/use/"
        >
          <span
            class="arrow"
          >
            →
          </span>
          Anwendung
        
        </router-link-stub>
      </h3>
       
      <ul>
        <li>
          <router-link-stub
            class="black"
            to="/de/use/#_1-eine-auf-ethereum-basierende-anwendung-verwenden"
          >
            
            Was kann ich schon heute mit Ethereum tun?
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/de/use/#_2-was-ist-eth-und-wie-bekomme-ich-es"
          >
            
            Wie komme ich an Ether?
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/de/use/#_3-was-ist-eine-wallet-und-welche-sollte-ich-verwenden"
          >
            
            Was ist eine Wallet?
          
          </router-link-stub>
        </li>
      </ul>
    </div>
     
    <div
      class="intro-block"
    >
      <h3>
        <router-link-stub
          to="/de/learn/"
        >
          <span
            class="arrow"
          >
            →
          </span>
          Lernen
        
        </router-link-stub>
      </h3>
       
      <ul>
        <li>
          <router-link-stub
            class="black"
            to="/de/learn/#ethereum-grundlagen"
          >
            
            Die Ethereum-Grundlagen
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/de/learn/#wie-ethereum-funktioniert"
          >
            
            Wie Ethereum funktioniert
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/de/learn/#eth-2-0"
          >
            
            ETH 2.0
          
          </router-link-stub>
        </li>
      </ul>
    </div>
     
    <div
      class="intro-block"
    >
      <h3>
        <router-link-stub
          to="/de/developers/"
        >
          <span
            class="arrow"
          >
            →
          </span>
          Entwickler
        
        </router-link-stub>
      </h3>
       
      <ul>
        <li>
          <router-link-stub
            class="black"
            to="/de/developers/#erste-schritte"
          >
            
            Erste Schritte
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/de/developers/#smart-contract-sprachen"
          >
            
            Lerne Smart Contracts zu programmieren
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/de/developers/#entwicklertools"
          >
            
            Finde die neuesten Entwickler-Tools
          
          </router-link-stub>
        </li>
      </ul>
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`HomePage show enterprise with en-US 1`] = `
<div>
  <div
    class="headline-text-container"
  >
    <h1
      class="headline-title"
    >
      Ethereum is a global, open-source platform for decentralized applications.
    </h1>
     
    <div
      class="headline-subtitle"
    >
      On Ethereum, you can write code that controls digital value, runs exactly as programmed, and is accessible anywhere in the world.
    </div>
  </div>
   
  <div
    class="intro-blocks"
  >
    <div
      class="intro-block"
    >
      <h3>
        <router-link-stub
          to="/beginners/"
        >
          <span
            class="arrow"
          >
            →
          </span>
          Beginners
        
        </router-link-stub>
      </h3>
       
      <ul>
        <li
          class="highlight"
        >
          <router-link-stub
            class="black"
            to="/beginners/"
          >
            
            Completely new to Ethereum?
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/beginners/"
          >
            
            What is Ethereum?
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/beginners/"
          >
            
            Why should I care?
          
          </router-link-stub>
        </li>
      </ul>
    </div>
     
    <div
      class="intro-block"
    >
      <h3>
        <router-link-stub
          to="/use/"
        >
          <span
            class="arrow"
          >
            →
          </span>
          Use
        
        </router-link-stub>
      </h3>
       
      <ul>
        <li>
          <router-link-stub
            class="black"
            to="/use/#_1-use-an-application-built-on-ethereum"
          >
            
            What can I do with Ethereum today?
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/use/#_2-what-is-eth-and-how-do-i-get-it"
          >
            
            How do I get Ether?
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/use/#_3-what-is-a-wallet-and-which-one-should-i-use"
          >
            
            What's a wallet?
          
          </router-link-stub>
        </li>
      </ul>
    </div>
     
    <div
      class="intro-block"
    >
      <h3>
        <router-link-stub
          to="/learn/"
        >
          <span
            class="arrow"
          >
            →
          </span>
          Learn
        
        </router-link-stub>
      </h3>
       
      <ul>
        <li>
          <router-link-stub
            class="black"
            to="/learn/#ethereum-basics"
          >
            
            Beginner resources
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/learn/#how-ethereum-works"
          >
            
            How Ethereum works
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/learn/#eth-2-0"
          >
            
            ETH 2.0
          
          </router-link-stub>
        </li>
      </ul>
    </div>
     
    <div
      class="intro-block"
    >
      <h3>
        <router-link-stub
          to="/developers/"
        >
          <span
            class="arrow"
          >
            →
          </span>
          Developers
        
        </router-link-stub>
      </h3>
       
      <ul>
        <li>
          <router-link-stub
            class="black"
            to="/developers/#getting-started"
          >
            
            Getting started guides
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/developers/#smart-contract-languages"
          >
            
            Learn to program smart contracts
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/developers/#developer-tools"
          >
            
            Find the latest developer tools
          
          </router-link-stub>
        </li>
      </ul>
    </div>
     
    <div
      class="intro-block"
    >
      <h3>
        <router-link-stub
          to="/enterprise/"
        >
          <span
            class="arrow"
          >
            →
          </span>
          Enterprise
        
        </router-link-stub>
      </h3>
       
      <ul>
        <li>
          <router-link-stub
            class="black"
            to="/enterprise/#why-enterprise-ethereum"
          >
            
            Why Enterprise Ethereum?
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/enterprise/#enterprise-features"
          >
            
            Enterprise Features
          
          </router-link-stub>
        </li>
         
        <li>
          <router-link-stub
            class="black"
            to="/enterprise/#enterprise-developer-community"
          >
            
            Enterprise Developer Community
          
          </router-link-stub>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
